python/core/auto_generated/qgsaggregatecalculator.sip.in

```diff
@@ -120,7 +120,8 @@ Returns the delimiter used for joining values with the StringConcatenate aggrega
 %End
 
     QVariant calculate( Aggregate aggregate, const QString &fieldOrExpression,
-                        QgsExpressionContext *context = 0, bool *ok = 0 ) const;
+                        QgsExpressionContext *context = 0, bool *ok = 0,
+                        const QgsFeatureIds *ids = 0) ) const;
```

Comment:
How about a QgsFeatureRequest instead of the ids here?

Reply:
No, the bigger PR is dependent on this. If needed someone could overload the function but in order to perform calculations on every feature associated with a given symbol I need to know which feature is associated with which symbol. To have access to this this list of symbol is stored in the featurecounter and is provided by the symbol expression.

Yes this is the part that causes segfault but I don't see any workaround to properly link the symbols to their corresponding features and passing it down to the calculator.

The feature ids are passed to the featurerequest used in the aggregator.

Reply:
@m-kuhn if we wanted to provide a feature request the optimal thing would be to leave the head of this function as a generator for a QgsFeatureRequest and move the code of the core to another function that would directly use the qgsFeatureRequest, similar to what Mr. Dawson did where he just split the function into its internal part and made front-facing code that would allow users to provide their inputs for iteration.

But this is out of the scope of this PR.

Reply:
Internally, the ids are fed into a QgsFeatureRequest. I think it should be trivial to start from a request that is provided as a parameter instead of a new one there. Did I overlook something?

Reply:
Yes, a feature request is created already in the original implementation, all this does is use the proper parameter of the featurerequest to filter out unwanted features, this does not add a new featurerequest. Also inputing directly a feature request might be less api/user friendly as this is already handled internally in the function. We'll go over that once we get to talk.